src/plugin/vectortile/vectortileplugin.js

```diff
@@ -0,0 +1,72 @@
+goog.module('plugin.vectortile.VectorTilePlugin');
+goog.module.declareLegacyNamespace();
+
+const MapContainer = goog.require('os.MapContainer');
+const OLDoubleClick = goog.require('os.interaction.DoubleClick');
+const LayerConfigManager = goog.require('os.layer.config.LayerConfigManager');
+const AbstractPlugin = goog.require('os.plugin.AbstractPlugin');
+const {ID} = goog.require('plugin.vectortile');
+const DoubleClick = goog.require('plugin.vectortile.DoubleClick');
+const VectorTileLayerConfig = goog.require('plugin.vectortile.VectorTileLayerConfig');
+
+
+/**
+ * Plugin to add vector tile support to OpenSphere.
+ */
+class VectorTilePlugin extends AbstractPlugin {
+  /**
+   * Constructor.
+   */
+  constructor() {
+    super();
+    this.id = ID;
+  }
+
+  /**
+   * @inheritDoc
+   */
+  init() {
+    // TODO: Add provider/descriptor? Currently only configurable with config descriptors.
+    // TODO: Add import UI so users can add vector tile layers.
+
+    const lcm = LayerConfigManager.getInstance();
+    lcm.registerLayerConfig(ID, VectorTileLayerConfig);
+
+    const map = MapContainer.getInstance().getMap();
+    const interactions = map.getInteractions();
+
+    // find the feature double click interaction
+    const arr = interactions.getArray();
+    let i = arr.length;
+    while (i--) {
+      if (arr[i] instanceof OLDoubleClick) {
+        break;
+      }
+    }
+
+    // ... and add ours just before it
+    if (i > -1) {
+      interactions.insertAt(i, new DoubleClick());
```

Comment:
Any concern with this mucking with other interactions? Performance concerns? Drill pick is pretty pretty pretty slow already with a large number of stacked features.

Reply:
Interactions are run in reverse order, so this is running the VT double click after the normal one. If features are hit, this won't run. If features aren't hit, it will run the VT double click.

Reply:
I did notice the hover interaction was causing performance issues with a vector tile layer, because the hit detection done by `forEachFeatureAtPixel` is replaying the tile render. Vector tile layers aren't supported by hover/select/etc so I omitted them in `os.interaction.defaultLayerFilter`.